### PR TITLE
Feature/612

### PR DESCRIPTION
**Pull request by yuliiakucher**
Created new methods for resending email and otp-code sendind. Created new pug-page for otp-code. 

image

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date front-end and admin part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: modules/business-text/business-text.service.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ class BusinessTextService {
 
   async updateBusinessText(id, businessText, files) {
     const pages = await this.checkBusinessTextExistByCode(businessText);
+    console.log('pages', pages);
```
Comment (AndriyLymych) [inline modules/business-text/business-text.service.js:37]: remove console.log

File: modules/business-text/business-text.service.js [hunk 2]
```diff
@@ -105,7 +106,8 @@ class BusinessTextService {
   }
 
   async checkBusinessTextExistByCode(data) {
-    return await BusinessText.find({ code: data.code }).exec();
+    console.log(BusinessText.find({ code: data.code }).exec());
```
Comment (AndriyLymych) [inline modules/business-text/business-text.service.js:109]: same

File: modules/user/user.graphql.js [hunk 3]
```diff
@@ -147,8 +147,22 @@ const adminRegisterInput = `
 input AdminRegisterInput {
     email: String!
     role: String!
+    code: String
```
Comment (AndriyLymych) [inline modules/user/user.graphql.js:150]: rename prop

File: modules/user/user.resolver.js [hunk 4]
```diff
@@ -101,9 +101,29 @@ const userMutation = {
       };
     }
   },
-  registerAdmin: async (parent, args) => {
+  registerAdmin: async (parent, args, context) => {
```
Comment (AndriyLymych) [inline modules/user/user.resolver.js:104]: pretify

File: modules/user/user.resolver.js [hunk 5]
```diff
@@ -101,9 +101,29 @@ const userMutation = {
...
     try {
-      return await userService.registerAdmin(args.user);
+      return await userService.registerAdmin(args.user, context.user);
+    } catch (err) {
+      return {
+        statusCode: BAD_REQUEST,
+        message: err.message,
+      };
+    }
```
Comment (AndriyLymych) [inline modules/user/user.resolver.js:112]: use RUlE  ERROR PLEASE!

File: modules/user/user.resolver.js [hunk 6]
```diff
@@ -101,9 +101,29 @@ const userMutation = {
...
+  },
+  resendEmailToConfirmAdmin: async (parent, args) => {
+    try {
+      return await userService.resendEmailToConfirmAdmin(args.user);
+    } catch (err) {
+      return {
+        statusCode: BAD_REQUEST,
+        message: err.message,
+      };
+    }
```
Comment (AndriyLymych) [inline modules/user/user.resolver.js:122]: same

File: modules/user/user.resolver.js [hunk 7]
```diff
@@ -101,9 +101,29 @@ const userMutation = {
...
+  },
+  confirmSuperadminCreation: async (parent, args) => {
```
Comment (AndriyLymych) [inline modules/user/user.resolver.js:124]: same

File: modules/user/user.service.js [hunk 8]
```diff
@@ -622,11 +631,15 @@ class UserService extends FilterHelper {
     return true;
   }
 
-  async registerAdmin(userInput) {
-    const { email, role } = userInput;
+  async registerAdmin({ email, role, code }, admin) {
+    if (code && code !== admin.otp_code) {
+      throw new RuleError(INVALID_CODE, FORBIDDEN);
```
Comment (AndriyLymych) [inline modules/user/user.service.js:636]: set more correct error name

File: modules/user/user.service.js [hunk 9]
```diff
@@ -64,6 +72,7 @@ const RuleError = require('../../errors/rule.error');
 const {
   roles: { ADMIN, SUPERADMIN },
 } = require('../../consts/');
+const { generateOrderNumber } = require('../../utils/order.utils');
```
Comment (AndriyLymych) [inline modules/user/user.service.js:75]: use alias

File: modules/user/user.service.js [hunk 10]
```diff
@@ -644,43 +657,104 @@ class UserService extends FilterHelper {
...
+  }
+
+  async resendEmailToConfirmAdmin({ email }) {
+    const isAdminExists = await User.findOne({ email }).exec();
+
+    if (!isAdminExists) {
+      throw new RuleError(USER_NOT_FOUND, NOT_FOUND);
+    }
+
+    const { accesToken } = generateTokens(isAdminExists._id, {
```
Comment (AndriyLymych) [inline modules/user/user.service.js:712]: use alias for correct naming

File: modules/user/user.service.js [hunk 11]
```diff
@@ -644,43 +657,104 @@ class UserService extends FilterHelper {
...
+      token: invitationalToken,
+    });
+
+    return { isSuccess: true };
+  }
+
   async completeAdminRegister(updatedUser, token) {
-    const { firstName, lastName, password } = updatedUser;
-    let decoded;
+    const { password } = updatedUser;
```
Comment (AndriyLymych) [inline modules/user/user.service.js:726]: use destructurization in func params

**Review by AndriyLymych — DISMISSED** [review]

**Review by AndriyLymych — APPROVED** [review]

**Review by twihell — APPROVED** [review]
Thanks!

**Review by fulljsdev — APPROVED** [review]

**Pull request merged**